registry api implementation

kkoralsky opened the pull request:
this is for #723. 

Im willing to add documentation too

gpocentek commented:
Hi @kkoralsky

Thank you for this patch! Would you mind rebasing your change? That will help re-running the CI tests too.

Thanks!

kkoralsky commented:
@gpocentek i've rebased the PR and added some docs. this should be done, although i encourge to merge #760 as well to make: `gitlab project-registry-tag delete_in_bulk --project-id x --repository-id y` functional

kkoralsky commented on docs/gl_objects/repositories.rst:
```diff
@@ -0,0 +1,28 @@
+
+Get the list of container registry repositories associated with the project::
+
+      registries = project.registries.list()
```
ah, yes this should be repository/repositories. I've replaced that

max-wittig commented:
Thanks for your contribution!
